instance/models/mixins/openedx_config.py

```diff
@@ -453,6 +461,14 @@ def _get_configuration_variables(self):
             template["EDXAPP_CMS_ENV_EXTRA"]["ADDL_INSTALLED_APPS"].append(hb_app)
             template["EDXAPP_LMS_ENV_EXTRA"]["ADDL_INSTALLED_APPS"].append(hb_app)
 
+        # See https://github.com/edx/cs_comments_service/pull/323
+        if self._is_openedx_release_in(['ironwood']):
+            template["forum_source_repo"] = "https://github.com/open-craft/cs_comments_service.git"
+            template["forum_version"] = "opencraft-release/ironwood.2"
+        if self._is_openedx_release_in(['juniper']):
+            template["forum_source_repo"] = "https://github.com/open-craft/cs_comments_service.git"
+            template["FORUM_VERSION"] = "opencraft-release/juniper.2"
+
```

Comment:
@swalladge quality failure 😦 

> pylint: too-many-branches / Too many branches

Reply:
:joy:

Reply:
This shouldn't be hard-coded to OpenCraft branches, if we still expect that other people use Ocim to manage their instances. But I don't know how to override this branch. Couldn't we modify our playbooks instead? (Not sure).

Right now the `openedx_release` field [is used as default for the forum, notifier, etc.](https://github.com/open-craft/opencraft/blob/734bac60816da3796c1e99e504ffd767279b9f5a/instance/models/openedx_appserver.py#L84).
An opposite way would be to override `forum_version` through `extra_configuration_settings` in every instance we create. Manually.
I don't know whether we have any option to automatically apply some temporary variables specific to our instances. Something like a default `configuration_extra_settings`, that makes it clear that it is overriding the defaults, while we keep simple/sane defaults. This isn't trivial. 

So, apart from modifying the playbooks instead, or doing the `forum_version` override manually in each instance, or finding another way to override defaults… Another way to do this would be to do this hard-coded override *only* for instances built off OpenCraft branches (adding a condition to the `if` here). But we'd still be hard-coding OpenCraft and OpenCraft's forks in the code.

Reply:
> Something like a default configuration_extra_settings,

that is what this openedx_config is supposed to be - a safe, tested set of defaults for instances. I think this is fine, because if anyone else uses ocim, they can add their own extra settings overrides.

Reply:
I don't object to merging it; I just think it's bad practice to hardcode OpenCraft-specific logic or branches in the code. It creates more possible bugs (e.g. if edX releases juniper.3 and we deploy it, this temporary snippet would force juniper.2 for the forum instead of auto-using juniper.3), it creates more maintenance work, …
There should be at least some comment asking to remove this temporary code.

Other ways, like creating a new setting to specify FORUM_VERSION, are more complex.

Reply:
@clemente I agree it's bad practice, so have added a TODO to remind us to fix this when/if we can.

But since this fix is breaking juniper.2 redeployments and new self-hosted services, and this code fix isn't available in any upstream branches yet, I think it's all we can do right now.

Plus, even once https://github.com/edx/cs_comments_service/pull/323 is ported to `juniper`, it'll only be on `juniper.master`, so we can't just use our usual [`openedx_release`](https://github.com/open-craft/opencraft/blob/7ff66c37fcd4dd52486f54772081a807281a9fa0/instance/models/mixins/openedx_config.py#L186-L187) for forums either while we're on `juniper.2`.

A better long-term approach might be to add additional `FORUM_STABLE_REF` and `FORUM_STABLE_REPO` environment variables, and configure it for our deployment, but as noted, anyone trying to use Ocim with `juniper.2` or `ironwood.2` right now will be unable to provision.